src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,432 @@
+        * xlabel -> passed to :func:`~arviz_plots.visuals.labelled_x`
+        * ylabel -> passed to :func:`~arviz_plots.visuals.labelled_y`
+        * mean -> passed to :func:`~arviz.plots.visuals.line_xy`
+        * sd -> passed to :func:`~arviz.plots.visuals.line_xy`
+        * min_ess -> passed to :func:`~arviz.plots.visuals.line_xy`
+
+    stats_kwargs : mapping, optional
+        Valid keys are:
+
+        * ess -> passed to ess, method = 'local' or 'quantile' based on `kind`
+        * mean -> passed to ess, method='mean'
+        * sd -> passed to ess, method='sd'
+
+    pc_kwargs : mapping
+        Passed to :class:`arviz_plots.PlotCollection.wrap`
+
+    Returns
+    -------
+    PlotCollection
+    """
+    # initial defaults
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+
+    # mutable inputs
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+
+    # processing dt/group/coords/filtering
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # set plot collection initialization defaults if it doesnt exist
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        if "chain" in distribution:
+            pc_kwargs["aes"].setdefault("overlay", ["chain"])
+        if "model" in distribution:
+            pc_kwargs["aes"].setdefault("color", ["model"])
+            n_models = distribution.sizes["model"]
+            x_diff = min(1 / n_points / 3, 1 / n_points * n_models / 10)
+            pc_kwargs.setdefault("x", np.linspace(-x_diff, x_diff, n_models))
+            pc_kwargs["aes"].setdefault("x", ["model"])
+        aux_dim_list = [dim for dim in pc_kwargs["cols"] if dim != "__variable__"]
+        plot_collection = PlotCollection.wrap(
+            distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    else:
+        aux_dim_list = list(
+            set(
+                dim for child in plot_collection.viz.children.values() for dim in child["plot"].dims
+            )
+        )
+
+    # set plot collection dependent defaults (like aesthetics mappings for each artist)
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault(kind, plot_collection.aes_set.difference({"overlay"}))
+    aes_map.setdefault("rug", {"overlay"})
```

Comment:
sd and mean should get the `color` aesthetic mapped to them, at least when `model` dimension is present. Otherwise you get n_model values of ess_mean for each plot which can't be used to plot a horizontal line at that y coordinate.

Moreover, if color is mapped to the model, then it can be used to distinguish the mean line from the sd line, they should get different linestyles, or there should be different aesthetic mappings defined depending on whether there are multiple models or not.